internal/metrics/metrics.go

```diff
@@ -0,0 +1,185 @@
+package metrics
+
+import (
+	"os"
+	"time"
+
+	gitopsSyncCommon "github.com/argoproj/gitops-engine/pkg/sync/common"
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	"github.com/numaproj-labs/numaplane/pkg/apis/numaplane/v1alpha1"
+)
+
+type MetricsServer struct {
+	syncCounter               *prometheus.CounterVec
+	gitSyncUpdateErrorCounter *prometheus.CounterVec
+	reconcileHistogram        *prometheus.HistogramVec
+	kubeRequestCounter        *prometheus.CounterVec
+	kubectlExecCounter        *prometheus.CounterVec
+	gitRequestCounter         *prometheus.CounterVec
+	gitRequestFailedCounter   *prometheus.CounterVec
+	gitRequestLatency         *prometheus.HistogramVec
+	monitoredKubeAPI          *prometheus.GaugeVec
+	kubeCachedResource        *prometheus.GaugeVec
+	kubeCacheFailed           *prometheus.CounterVec
+	hostname                  string
+}
+
+var (
+	descAppDefaultLabels = []string{"name", "namespace"}
+	syncCounter          = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "numaplane_app_sync_total",
```

Comment:
nit: change the name to `numaplane_gitsync_sync_total`, 'Number of GitSync syncs'.